OpenTrons/opentrons-api
# feat(app, step-generation, shared-data): FE support for moveToAddressableAreaForDropTip

closes RAUT-901

# Overview

Adds FE support for the new drop tip command which is used to drop tips in to the trash bin

# Test Plan

Currently blocked until there is API support for this. But:
1. create a PD protocol, either flex or OT-2 and create a transfer or mix step then drop tip into the trash bin, note: you can turn on the deck config ff if you want but its not necessary. 
2. open the JSON file and see that the command for drop tip includes "moveToAddressableAreaForDropTip" and does NOT include "dropTipInPlace" (since you're only dropping tip in the trash bin)
3. (This step won't work until we have backend support) upload the protocol to the app, in the command text there should be information about moving to the trash bin to drop tip.
# Changelog

- add new command to types
- add logic for command text and add to the test
- add a new atomic command in step generation
- add a robot state update fn to keep track of tip and liquid state
- plug new command into the movable trash commands util
- update all step-generation tests
- update the e2e tests

# Review requests

see test plan

# Risk assessment

low-ish
